contracts/src/v0.8/dev/VRFCoordinatorV2.sol

```diff
@@ -0,0 +1,527 @@
+import "../interfaces/VRFConsumerV2Interface.sol";
+
+contract VRFCoordinatorV2 is VRF, ConfirmedOwner, TypeAndVersionInterface {
+
+  LinkTokenInterface public immutable LINK;
+  AggregatorV3Interface public immutable LINK_ETH_FEED;
+  BlockHashStoreInterface public immutable BLOCKHASH_STORE;
+
+  error InsufficientBalance();
+  error InvalidConsumer(address consumer);
+  error InvalidNumberOfConsumers(uint256 have, uint16 want);
+  error InvalidSubscription();
+  error MustBeSubOwner();
+  struct Subscription {
+    uint96 balance; // Common link balance used for all consumer requests.
+    address owner; // Owner can fund/withdraw/cancel the sub
+    address[] consumers; // List of addresses which can consume using this subscription.
+  }
+  mapping(uint64 /* subId */ => Subscription /* subscription */) private s_subscriptions;
+  uint64 private currentSubId;
+  event SubscriptionCreated(uint64 subId, address owner, address[] consumers);
+  event SubscriptionFundsAdded(uint64 subId, uint256 oldBalance, uint256 newBalance);
+  event SubscriptionConsumersUpdated(uint64 subId, address[] oldConsumers, address[] newConsumers);
+  event SubscriptionFundsWithdrawn(uint64 subId, uint256 oldBalance, uint256 newBalance);
+  event SubscriptionCanceled(uint64 subId, address to, uint256 amount);
+
+  error RequestBlockConfsTooLow(uint64 have, uint64 want);
+  error UnregisteredKeyHash(bytes32 keyHash);
+  error KeyHashAlreadyRegistered(bytes32 keyHash);
+  error InvalidFeedResponse(uint256 linkWei);
+  error InsufficientGasForConsumer(uint256 have, uint256 want);
+  error InvalidProofLength(uint256 have, uint256 want);
+  error NoCorrespondingRequest();
+  error IncorrectCommitment();
+  error BlockHashNotInStore(uint256 blockNum);
+  // Just to relieve stack pressure
+  struct FulfillmentParams {
+    uint64 subId;
+    uint64 callbackGasLimit;
+    uint64 numWords;
+    address sender;
+  }
+  mapping(bytes32 /* keyHash */ => address /* oracle */) private s_serviceAgreements;
+  mapping(address /* oracle */ => uint96 /* LINK balance */) private s_withdrawableTokens;
+  mapping(bytes32 /* keyHash */ => mapping(address /* consumer */ => uint256 /* nonce */)) public s_nonces;
+  mapping(uint256 /* requestID */ => bytes32) private s_callbacks;
+  event NewServiceAgreement(bytes32 keyHash, address oracle);
+  event RandomWordsRequested(
+    bytes32 indexed keyHash,
+    uint256 preSeedAndRequestId,
+    uint64 subId,
+    uint64 minimumRequestConfirmations,
+    uint64 callbackGasLimit,
+    uint64 numWords,
+    address sender
+  );
+  event RandomWordsFulfilled(
+    uint256 requestId,
+    uint256[] output,
+    bool success
+  );
+
+  struct Config {
+    // Gas to cover oracle payment after we calculate the payment.
+    // We make it configurable in case those operations are repriced.
+    uint32 gasAfterPaymentCalculation;
+    uint32 stalenessSeconds;
+    uint16 minimumRequestBlockConfirmations;
+    uint16 maxConsumersPerSubscription;
```

Comment:
Before https://github.com/smartcontractkit/chainlink/pull/4526/commits/f531949a587c6cb6bb049dfb77b1632dfb96d170 there was some variance on gas costs with the number of consumers, but I've since resolved that. I think we don't need it anymore